bpfd/src/rpc.rs

```diff
@@ -248,7 +248,9 @@ impl Bpfd for BpfdLoader {
                         match r.data() {
                             // If filtering on `bpfd Only`, this program is of type Unsupported so skip
                             Err(_) => {
-                                if request.get_ref().bpfd_programs_only() {
+                                if request.get_ref().bpfd_programs_only()
```

Comment:
Hrm so a "non bpfd program" (i.e `.data()` is `Err(e)`) should never have any "bpfd metadata" right? that means `!request.get_ref().match_metadata.is_empty()` __should__ always be false here....

Granted with that logic setting match_metadata to a "non nil value" is the same as saying "give me bpfd_programs_only" so I'm alright with this. 

The only question I have is in the future will we want to add filterable default metadata k/v's to ALL programs at some point to allow users to list programs based on both bpfd AND kernel attributes? If so this logic would need to change back again.

Reply:
If we `add filterable default metadata k/v's to ALL programs at some point` this code has to change anyway, because there is no filtering being done on non-bpfd added programs.

My first pass was to add bpfd-only=true to the agent, but then I decided that wasn't correct because all the programs that were returned (all the non-bpfd) didn't match the given metadata filter, so the decided the fix should be in bpfd.